<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FrameRamManager</class>
 <widget class="QFrame" name="FrameRamManager">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1241</width>
    <height>721</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>10</pointsize>
    <bold>true</bold>
   </font>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3" stretch="8,0">
   <property name="spacing">
    <number>5</number>
   </property>
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>5</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>5</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>当前文件：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditFileName">
           <property name="minimumSize">
            <size>
             <width>600</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonReLoad">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="text">
            <string>重新载入</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableView" name="tableViewData">
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox1">
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>250</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <pointsize>10</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="title">
          <string>数据块</string>
         </property>
         <widget class="QPushButton" name="pushButtonLoad">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>100</width>
            <height>32</height>
           </rect>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="text">
           <string>打开文件...</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonSave">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>30</y>
            <width>100</width>
            <height>32</height>
           </rect>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="text">
           <string>保存到文件</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonWrite">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>220</y>
            <width>100</width>
            <height>32</height>
           </rect>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="text">
           <string>写入设备</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonRead">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>270</y>
            <width>100</width>
            <height>32</height>
           </rect>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="text">
           <string>从设备读取</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>150</y>
            <width>111</width>
            <height>21</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>起始地址(Hex):</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
         <widget class="QLineEdit" name="lineEditDataAddr">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>150</y>
            <width>100</width>
            <height>20</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>0000</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>180</y>
            <width>101</width>
            <height>21</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>读/写数量:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBoxDataCount">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>180</y>
            <width>101</width>
            <height>22</height>
           </rect>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>10000000</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
          <property name="displayIntegerBase">
           <number>10</number>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonGenData">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>220</y>
            <width>101</width>
            <height>32</height>
           </rect>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="text">
           <string>生成随机数据</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonClearData">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>270</y>
            <width>101</width>
            <height>32</height>
           </rect>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="text">
           <string>清空列表</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_4">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>80</y>
            <width>91</width>
            <height>20</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>D:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBoxD">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>80</y>
            <width>101</width>
            <height>22</height>
           </rect>
          </property>
          <property name="maximum">
           <number>1024</number>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBoxL">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>110</y>
            <width>101</width>
            <height>22</height>
           </rect>
          </property>
          <property name="maximum">
           <number>64</number>
          </property>
          <property name="value">
           <number>13</number>
          </property>
         </widget>
         <widget class="QLabel" name="label_5">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>110</y>
            <width>91</width>
            <height>20</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>左移:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QTableView" name="tableViewFile">
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="alternatingRowColors">
        <bool>false</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox2">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>250</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="title">
        <string>文件操作</string>
       </property>
       <widget class="QPushButton" name="pushButtonAddFile">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>30</y>
          <width>100</width>
          <height>32</height>
         </rect>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="text">
         <string>添加文件...</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButtonRemoveFile">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>70</y>
          <width>100</width>
          <height>32</height>
         </rect>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="text">
         <string>从列表移除</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButtonWriteFile">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>110</y>
          <width>100</width>
          <height>32</height>
         </rect>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="text">
         <string>写入设备</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButtonOpenFile">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>110</y>
          <width>100</width>
          <height>32</height>
         </rect>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="text">
         <string>打开文件</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBoxHexFile">
        <property name="geometry">
         <rect>
          <x>25</x>
          <y>30</y>
          <width>80</width>
          <height>22</height>
         </rect>
        </property>
        <property name="text">
         <string>HEX 格式</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>5</x>
          <y>58</y>
          <width>51</width>
          <height>22</height>
         </rect>
        </property>
        <property name="text">
         <string>行数：</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBoxLineCount">
        <property name="geometry">
         <rect>
          <x>55</x>
          <y>58</y>
          <width>65</width>
          <height>20</height>
         </rect>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>5</x>
          <y>85</y>
          <width>51</width>
          <height>22</height>
         </rect>
        </property>
        <property name="text">
         <string>地址：</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEditFileAddr">
        <property name="geometry">
         <rect>
          <x>55</x>
          <y>85</y>
          <width>65</width>
          <height>20</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>0000</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>160</y>
          <width>211</width>
          <height>16</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(255, 0, 0);</string>
        </property>
        <property name="text">
         <string>注意：行数为 0 表示整个文件</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
